### PR TITLE
release: Build binaries in container

### PR DESCRIPTION
**Pull request by serathius**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
+FROM golang:1.17.8 as build
```
Comment (mrueg) [inline Dockerfile:1]: do you want to upgrade to 1.17.9?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+RUN mkdir -p /go/go.etcd.io/etcd
+WORKDIR /go/go.etcd.io/etcd
+ADD go.mod go.mod
```
Comment (mrueg) [inline Dockerfile:4]: In general COPY would be better: https://github.com/hadolint/hadolint/wiki/DL3020

File: scripts/build-release.sh [hunk 3]
```diff
@@ -18,14 +18,53 @@ if ! command -v docker >/dev/null; then
...
+    for GOARCH in "${TARGET_ARCHS[@]}"; do
+      IMAGE="gcr.io/etcd-development/etcd:${VERSION}-${GOOS}-${GOARCH}"
+      if [ "${GOOS}" == "linux" ]; then
+        IMAGE="gcr.io/etcd-development/etcd:${VERSION}-${GOARCH}"
+      fi
 
-  for TARGET_ARCH in "amd64" "arm64" "ppc64le" "s390x"; do
-    log_callout "Building ${TARGET_ARCH} docker image..."
-    GOOS=linux GOARCH=${TARGET_ARCH} BINARYDIR=release/etcd-${VERSION}-linux-${TARGET_ARCH} BUILDDIR=release ./scripts/build-docker.sh "${VERSION}"
+      docker build -t "${IMAGE}" --build-arg GOARCH=${GOARCH} --build-arg GOOS=${GOOS} .
```
Comment (vimalk78) [inline scripts/build-release.sh:67]: where is the image being pushed?

File: scripts/build-release.sh [hunk 4]
```diff
@@ -18,14 +18,53 @@ if ! command -v docker >/dev/null; then
...
+    TARGET_ARCHS=("amd64")
+
+    if [ ${GOOS} == "linux" ]; then
+      TARGET_ARCHS+=("arm64")
+      TARGET_ARCHS+=("ppc64le")
+      TARGET_ARCHS+=("s390x")
+    fi
+
+    if [ ${GOOS} == "darwin" ]; then
+      TARGET_ARCHS+=("arm64")
```
Comment (camilamacedo86) [inline scripts/build-release.sh:58]: Thank you :-)

**Pull request closed without merging**